python/ray/train/lightgbm/lightgbm_checkpoint.py

```diff
@@ -54,14 +53,16 @@ def from_model(
             >>>
             >>> predictor = LightGBMPredictor.from_checkpoint(checkpoint)  # doctest: +SKIP # noqa: #501
         """
-        booster.save_model(os.path.join(path, MODEL_KEY))
```

Comment:
wouldn't this lead to saving full checkpoint to a new temp directory rather than a user given path ? Seems current behavior is better that user has control over where they want to save preprocessor to, and this PR's change is changing it to an ephemeral path we created

Reply:
In this proposal, the user doesn't supply a path anymore.

This is a simple API that solves a lot of headaches (e.g. user managing temporary directories). If users need more efficiency by specifying their own non-ephemeral path, we can add this if this is requested. Most xgboost models are small, the biggest ones I've seen in production are about 50MB. So ser/de should be relatively fast.